plugins/gradle/src/org/jetbrains/plugins/gradle/service/project/GradleProjectResolverUtil.java

```diff
@@ -604,8 +604,7 @@ else if (ownerDataNode.getData() instanceof DependencyData) {
         final ExternalProjectDependency projectDependency = (ExternalProjectDependency)mergedDependency;
 
         Collection<ProjectDependencyInfo> projectDependencyInfos = new ArrayList<>();
-        String selectionReason = projectDependency.getSelectionReason();
-        if ("composite build substitution".equals(selectionReason) && resolverCtx.getSettings() != null) {
+        if (resolverCtx.getSettings() != null) {
```

Comment:
This will break another kind of dependency resolution, because it will ignore `sourceSetMap` and `artifactsMap` mappings.